apple/internal/watchos_rules.bzl

```diff
@@ -592,7 +592,11 @@ def _watchos_extension_impl(ctx):
     # TODO(min(Xcode) >= 11): Make this unconditional when the minimum supported Xcode is Xcode 11.
     xcode_config = ctx.attr._xcode_config[apple_common.XcodeVersionConfig]
     if xcode_support.is_xcode_at_least_version(xcode_config, "11"):
-        extra_linkopts = ["-e", "_WKExtensionMain"]
+        # watchOS application extensions have a different entry point.
+        if ctx.attr.application_extension:
```

Comment:
We also need to update all `product_type` references in here to use `apple_product_type.app_extension` instead of `rule_descriptor.product_type` when this is set.